File: 881-boats-to-save-people/881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        low=0
        ans=0
        high=len(people)-1
        while(high>=low):
            if (people[low]+people[high]<=limit):
                low+=1
            high-=1
            ans+=1
        return ans
        

File: 881-boats-to-save-people/881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(),people.end()); //sort the vector
        int i=0,j=people.size()-1; //i for light person and j for heavy person
        int cnt=0; //cnt is zero
        while(i<=j)
        {
            if(people[i]+people[j]<=limit)
            {
                i++;
                j--;
            }
            else
            {
                j--;
            }
            cnt++;
        }
        return cnt;        
    }
};